File: es6/prod/src/web_workers/ui/router_providers.js
import { MessageBasedPlatformLocation } from './platform_location';
import { CONST_EXPR } from 'angular2/src/facade/lang';
import { BrowserPlatformLocation } from 'angular2/src/router/browser_platform_location';
import { APP_INITIALIZER, Provider, Injector, NgZone } from 'angular2/core';
export const WORKER_RENDER_ROUTER = CONST_EXPR([
    MessageBasedPlatformLocation,
    BrowserPlatformLocation,
    CONST_EXPR(new Provider(APP_INITIALIZER, { useFactory: initRouterListeners, multi: true, deps: CONST_EXPR([Injector]) }))
]);
function initRouterListeners(injector) {
    return () => {
        let zone = injector.get(NgZone);
        zone.run(() => injector.get(MessageBasedPlatformLocation).start());
    };
}

File: lib/src/web_workers/ui/router_providers.dart
library angular2.src.web_workers.ui.router_providers;

import "platform_location.dart" show MessageBasedPlatformLocation;
import "package:angular2/src/router/browser_platform_location.dart"
    show BrowserPlatformLocation;
import "package:angular2/core.dart"
    show APP_INITIALIZER, Provider, Injector, NgZone;

const WORKER_RENDER_ROUTER = const [
  MessageBasedPlatformLocation,
  BrowserPlatformLocation,
  const Provider(APP_INITIALIZER,
      useFactory: initRouterListeners, multi: true, deps: const [Injector])
];
dynamic /* () => void */ initRouterListeners(Injector injector) {
  return () {
    var zone = injector.get(NgZone);
    zone.run(() => injector.get(MessageBasedPlatformLocation).start());
  };
}
